Realm/ObjectServerTests/SwiftPartialSyncTests.swift

```diff
@@ -0,0 +1,365 @@
+            XCTAssertEqual(object.string, "partial")
+        }
+
+        let results2 = realm.objects(SwiftPartialSyncObjectA.self).sorted(byKeyPath: "number", ascending: false)
+        waitForState(results2.subscribe(named: "query2", limit: 2), .complete)
+        XCTAssertEqual(results2.count, 3)
+        XCTAssertEqual(realm.objects(SwiftPartialSyncObjectA.self).count, 3)
+        for object in results2 {
+            XCTAssertTrue(object.number == 6 || object.number >= 8,
+                          "\(object.number) == 6 || \(object.number) >= 8")
+            XCTAssertEqual(object.string, "partial")
+        }
+
+        waitForState(results2.subscribe(named: "query2", limit: 1, update: true), .complete)
+        XCTAssertEqual(results2.count, 2)
+        XCTAssertEqual(realm.objects(SwiftPartialSyncObjectA.self).count, 2)
+        for object in results2 {
+            XCTAssertTrue(object.number == 6 || object.number == 9,
+                          "\(object.number) == 6 || \(object.number) == 9")
+            XCTAssertEqual(object.string, "partial")
+        }
+    }
+
+    func testPartialSyncSubscriptions() {
+        let credentials = SyncCredentials.usernamePassword(username: #function, password: "a", register: true)
+        let user = try! synchronouslyLogInUser(for: credentials, server: authURL)
+        let realm = try! synchronouslyOpenRealm(configuration: user.configuration())
+
+        XCTAssertEqual(realm.subscriptions().count, 0)
+        XCTAssertNil(realm.subscription(named: "query"))
+
+        let subscription = realm.objects(SwiftPartialSyncObjectA.self).filter("number > 5").subscribe(named: "query")
+        XCTAssertEqual(realm.subscriptions().count, 0)
+        XCTAssertNil(realm.subscription(named: "query"))
+        waitForState(subscription, .complete)
+
+        XCTAssertEqual(realm.subscriptions().count, 1)
+        let sub2 = realm.subscriptions().first!
+        XCTAssertEqual(sub2.name, "query")
+        XCTAssertEqual(sub2.state, .complete)
+        let sub3 = realm.subscription(named: "query")!
+        XCTAssertEqual(sub3.name, "query")
+        XCTAssertEqual(sub3.state, .complete)
+        for sub in realm.subscriptions() {
+            XCTAssertEqual(sub.name, "query")
+            XCTAssertEqual(sub.state, .complete)
+        }
+
+        XCTAssertNil(realm.subscription(named: "not query"))
+    }
+
+    func testSubscriptionPropertyUpdating() {
+        let credentials = SyncCredentials.usernamePassword(username: #function, password: "a", register: true)
+        let user = try! synchronouslyLogInUser(for: credentials, server: authURL)
+        let realm = try! synchronouslyOpenRealm(configuration: user.configuration())
+
+        // Create the initial subscription
+        let objects = realm.objects(SwiftPartialSyncObjectA.self)
+        let sub1 = objects.filter("number > 5").subscribe(named: "query")
+        XCTAssertEqual(sub1.name, "query")
+        XCTAssertEqual(sub1.query, "")
```

Comment:
The query string is constructed as part of creating a subscription from a Results, and until the subscription is written to the Realm doesn't exist anywhere except for in a local variable captured by a lambda. Reporting it accurately here would require either reconstructing the query string from the Results a second time or restructuring the flow around creating subscriptions to make it available earlier, and neither of those seemed worth it.

Maybe it'd be better to report `<pending>` or something until the actual query is available?

Reply:
I wouldn't expect it being expensive to reconstruct the query,  although I guess it can be in you e.g. use a very big IN query.

Reporting `<pending>` or `nill` both seems better options to me than the empty string though.